### DISCUSSION TITLE
Porous Flow - 0 liquid saturation?

### DISCUSSION THREAD

**Discussion by s-dunnim** [idaholab/moose#20862] [Q&A Modules: General] [answered]
I'm building a model of liquid being injected into a dry porous skeleton. I tried a formulating a single phase, unsaturated model, broadly following the steps in [page 10 of the tutorial](https://mooseframework.inl.gov/modules/porous_flow/tutorial_10.html) (I also tried a 2 phase model along the lines of page 11).

I would like the liquid saturation to start at 0, and increase as liquid is injected and flows throughout the skeleton. However in the [capillary pressure documentation](https://mooseframework.inl.gov/modules/porous_flow/capillary_pressure.html), it says "physically realistic problems rarely explore the Seff∼0 region". It is also clear from the equations given, and the models I have attempted, that 0 saturation is not really possible using the basic Porous Flow functionality.

Is there a way to get around this using the Porous Flow module, and start at 0 (liquid) saturation?
TIA

<details>
<summary>Single phase unsaturated input file for reference</summary>
<br>

```
[Mesh]
    [plate]
        type = GeneratedMeshGenerator
        dim = 2
        xmin = 0
        xmax = 1
        nx = 50
        ymin = 0
        ymax = 0.56
        ny = 28
    []
[]

[UserObjects]
    [dictator]
        type = PorousFlowDictator
        porous_flow_vars = porepressure
        number_fluid_phases = 1
        number_fluid_components = 1
    []
    [pc]
        type = PorousFlowCapillaryPressureVG
        alpha = 1E-6
        m = 0.6
    []
[]

[GlobalParams]
    PorousFlowDictator = dictator
    gravity = '0 0 0'
[]

[Variables]
    [porepressure]
        initial_condition = 0
    []
[]

[Kernels]
    [dmdt]
        type = PorousFlowMassTimeDerivative
        variable = porepressure
    []
    [flux]
        type = PorousFlowAdvectiveFlux
        variable = porepressure
    []
[]

[AuxVariables]
    [saturation]
        family = MONOMIAL
        order = CONSTANT
    []
[]

[AuxKernels]
    [saturation]
        type = PorousFlowPropertyAux
        variable = saturation
        property = saturation
    []
[]

[BCs]
    [injection]
        type = PorousFlowSink
        variable = porepressure
        boundary = left
        flux_function = -10
        fluid_phase = 0
        use_relperm = true
    []
[]

[Modules]
    [FluidProperties]
        [liquid]
            type = SimpleFluidProperties
            bulk_modulus = 2E9
            viscosity = 1E-3
            density0 = 1000
        []
    []
[]

[Materials]
    [porosity]
        type = PorousFlowPorosity
        porosity_zero = 0.1
    []
    [permeability]
        type = PorousFlowPermeabilityConst
        permeability = '1E-14 0 0   0 1E-14 0   0 0 1E-14'
    []
    [saturation_calculator]
        type = PorousFlow1PhaseP
        porepressure = porepressure
        capillary_pressure = pc
    []
    [temperature]
        type = PorousFlowTemperature
        temperature = 293
    []
    [massfrac]
        type = PorousFlowMassFraction
    []
    [liquid]
        type = PorousFlowSingleComponentFluid
        fp = liquid
        phase = 0
    []
    [relperm]
        type = PorousFlowRelativePermeabilityCorey
        n = 3
        s_res = 0.1
        sum_s_res = 0.1
        phase = 0
    []
[]

[Preconditioning]
    [smp]
        type = SMP
        full = true
        petsc_options_iname = '-pc_type -pc_factor_mat_solver_package'
        petsc_options_value = ' lu       mumps'
    []
[]

[Executioner]
    type = Transient
    solve_type = 'NEWTON'
    
    end_time = 10
    dt = 1

    automatic_scaling = true
[]

[Outputs]
    exodus = true
    print_linear_residuals = false
[]
```

</details>

**Comment by WilkAndy** [comment, marked as answer]
You are correct: liquid saturation = 0 will cause all sorts of headaches for you.  By carefully choosing Variables and capillary curves and relative-permeability curves, I believe you should be able to do it using 2-phase or 1-phase physics, but convergence may be a problem.  Before exploring that (it will be painful) please carefully consider whether zero saturation is truly correct.  Specifically, what experimental evidence do you have that your sample has zero saturation (I suppose zero saturation would require boiling the liquid off)?  If so, do you have a good feeling (experimental evidence) for your capillary pressure and relative-permeability end points, and shape around the end points, since they will crucially dictate the (numerical) behaviour?  Or, if you're not sure the liquid saturation is truly zero, can you start from the residual saturation instead?

**Reply by s-dunnim** [reply]
Thank you for the helpful reply. The liquid and porous skeleton in question are resin and a dry carbon fibre layup, so I can be sure the sample begins with 0 liquid saturation - although I suppose this isn't to say the residual saturation would be 0 once the resin is introduced. I don't have any data for the capillary pressure and relative permeability end points, and won't be able to determine this experimentally - the best I could do is review some literature for similar materials.

I will attempt the problem with some carefully selected curves as you suggest (presumably this will involve a log extension for the capillary pressure/saturation curve to allow the saturation to reach 0 - or one of the lower extensions used on the primary wetting curve in [hysteresis](https://mooseframework.inl.gov/modules/porous_flow/hysteresis.html)?). If that turns out to be unfeasible, I can just try a largely simplified custom implementation of Darcy flow...

**Reply by WilkAndy** [reply]
Yep, try the log extension.  We were pretty careful to plot representative curves in the documentation, so hopefully you should be able to choose something sensible.  I wouldn't try customising Darcy just yet.

**Reply by s-dunnim** [reply]
Thanks again for the advice.

**Reply by cpgr** [reply]
Interesting problem and one I suspect that we didn't even think about! Be sure to let us know how it goes.

**Reply by s-dunnim** [reply]
@cpgr I have had some limited success by setting the IC for porepressure to be negative (< -Pc_max for my chosen capillary pressure curve), which allows me to start at (close to) 0 saturation, and inject resin from a boundary using a PorousFlowSink. The results I'm getting look OK qualitatively but aren't exactly realistic... I've gone into more detail in a more [recent question](https://github.com/idaholab/moose/discussions/21000).
